### PR TITLE
Add group-user management workflow

### PR DESCRIPTION
**Pull request by pwalczysko**
Not ready for review yet - some formatting issues in the CLI section.

@jburel Is that repo okay to have this walkthrough ?

### CODE REVIEW

**Comment by jburel** [comment]
I think it is fine in this repo,
better to use group-user-management for the name of the file to match the data-management file

File: docs/group-user-management.rst [hunk 1]
```diff
@@ -0,0 +1,227 @@
+Administrate Groups and Users
+=============================
+
+Description:
+------------
+
+This chapter will show how to manage groups and users using graphical interface in OMERO.web and command line interface. Most of following tasks below can only be done by users with some
```
Comment (mtbc) [inline docs/group-user-management.rst]:
```suggestion
This chapter will show how to manage groups and users using the graphical interface in OMERO.web and the command-line interface. Most of following tasks below can only be done by users with some
```

File: docs/group-user-management.rst [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+
+Below are some useful links to know more about permissions and administrator privileges:
+
+-  https://docs.openmicroscopy.org/latest/omero/sysadmins/server-permissions.html
+
+-  https://docs.openmicroscopy.org/latest/omero/sysadmins/restricted-admins.html
+
+1. In your web browser, go to the server address provided.
+
+2. Login using the username and password provided.
```
Comment (mtbc) [inline docs/group-user-management.rst]:
```suggestion
2. Log in using the username and password provided.
```

File: docs/group-user-management.rst [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+
+      ii.  The Name and Permissions fields are mandatory.
+
+      iii. Click Save
+
+   d. To edit a Group:
+
+      iv.  Click on the Pencil button |image1|
+
+      v.   You can add or remove members or group’s owners. Change permissions etc..
```
Comment (mtbc) [inline docs/group-user-management.rst]:
```suggestion
      v.   You can add or remove members or group’s owners, change permissions etc..
```

File: docs/group-user-management.rst [hunk 4]
```diff
@@ -0,0 +1,227 @@
...
+
+5. Managing Users:
+
+   e. Click on the Users tab
+
+   f. You can search for users if desired
+
+   g. How to identify users:
+
+      viii. Users with administrators privileges with \ |image2|
```
Comment (mtbc) [inline docs/group-user-management.rst]:
```suggestion
      viii. Users with administrator privileges with \ |image2|
```
or a possessive apostrophe

File: docs/group-user-management.rst [hunk 5]
```diff
@@ -0,0 +1,227 @@
...
+
+      xiv.  You can select the role of the user
+
+      xv.   Click Save
+
+   i. To edit a User:
+
+      xvi.   Click on the Pencil button |image6|
+
+      xvii.  You can add or remove the user to/from a group. Modify the roles etc..
```
Comment (mtbc) [inline docs/group-user-management.rst]:
```suggestion
      xvii.  You can add or remove the user to/from a group, modify the roles etc..
```

File: docs/group-user-management.rst [hunk 6]
```diff
@@ -0,0 +1,227 @@
...
+
+      xviii. Click Save
+
+   j. Creating an administrator privileges allow to give some limited rights to some trusted users e.g. to allow a facility manager to import data for other users. It is currently preferable to create users with such role via the Web Interface.
```
Comment (mtbc) [inline docs/group-user-management.rst]:
```suggestion
   j. Creating an administrator with privileges allows to give some limited rights to some trusted users e.g. to allow a facility manager to import data for other users. It is currently preferable to create users with such roles via the Web Interface.
```
I think this is an improvement but not quite sure what you were going for so I'm just guessing.

File: docs/group-user-management.rst [hunk 7]
```diff
@@ -0,0 +1,227 @@
...
+
+6. Note that a user can manage his/her settings.
+
+   k. In the top-right corner of the webclient, click on your name
+
+   l. In the dropdown menu, click on User settings
+
+   m. In the dialog that pops up, the user can change password, default group i.e. the group he/she will log in by default etc.
```
Comment (mtbc) [inline docs/group-user-management.rst]:
```suggestion
   m. In the dialog that pops up, the user can change password, default group i.e. the group he/she will log in to by default etc.
```

File: docs/group-user-management.rst [hunk 8]
```diff
@@ -0,0 +1,227 @@
...
+
+       ``653 \| Lab1 \| rwra-- \| False \| 0 \| 0``
+
+    h. Change the group name to ``LabN``:
+
+       ``$ omero obj update ExperimenterGroup:653 name='LabN'``
+
+    i. Let’s reset the name back to ``Lab1`` to simplify the rest of the workflow
+
+    j.  Change groups permissions to read-write:
```
Comment (mtbc) [inline docs/group-user-management.rst]:
```suggestion
    j.  Change the group's permissions to read-write:
```

File: docs/group-user-management.rst [hunk 9]
```diff
@@ -0,0 +1,227 @@
...
+
+   e. Make a user inactive. User cannot be deleted but it is possible to prevent a user from logging in. For that we need to remove the user from the user group (internal OMERO group).
+
+      ``$ omero user leavegroup user --name=lpasteur``
+
+   f. To reactivate the user:
+
+      ``$ omero user joingroup user --name=lpasteur``
+
+   g. LDAP authentication. It is possible to convert non LDAP users to LDAP authentication using the command ``omero ldap setdn``. When using LDAP as an authentication backend, users when they log in will be added to the internal OMERO group called default unless they have already been added to a given group. To add a user before they have ever logged in to OMERO, first create the user (example user name is ``enoether``).
```
Comment (mtbc) [inline docs/group-user-management.rst]:
```suggestion
   g. LDAP authentication. It is possible to convert non-LDAP users to LDAP authentication using the command ``omero ldap setdn``. When using LDAP as an authentication backend, users when they log in will be added to the internal OMERO group called default unless they have already been added to a given group. To add a user before they have ever logged in to OMERO, first create the user (example user name is ``enoether``).
```

**Comment by mtbc** [comment]
Where literal user and group names are included in sentences it could be good to set them aside somehow with backticks or quotes or somesuch.

File: docs/group-user-management.rst [hunk 10]
```diff
@@ -0,0 +1,227 @@
...
+-----------------
+
+Administrate using the Web Interface
+------------------------------------
+
+Below are some useful links to know more about permissions and administrator privileges:
+
+-  https://docs.openmicroscopy.org/latest/omero/sysadmins/server-permissions.html
+
+-  https://docs.openmicroscopy.org/latest/omero/sysadmins/restricted-admins.html
```
Comment (will-moore) [inline docs/group-user-management.rst]: These links are repeated in the previous section. Bit confusing (user needs to check if they're the same links).

File: docs/group-user-management.rst [hunk 11]
```diff
@@ -0,0 +1,227 @@
...
+
+       ``$ omero group adduser user-1 --name=Lab1 --as-owner``
+
+    d. Let’s add trainer-1 as an owner of the group too:
+
+       ``$ omero group adduser trainer-1 --name=Lab1 --as-owner``
+
+    e. To remove user-1 from the list of owners (user-1 will still be a member of the Lab1 group):
+
+       ``$ omero user leavegroup Lab1 --name=user-1 --as-owner``
```
Comment (will-moore) [inline docs/group-user-management.rst]: If I did this without the ```--as-owner``` flag, would it remove the user-1 from the group?

File: docs/group-user-management.rst (hunk 9 above)
Comment (will-moore) [reply docs/group-user-management.rst]: How do you use ```omero ldap setdn```? e.g. ```omero ldap setdn --name enoether ``` ?

Comment (pwalczysko) [reply docs/group-user-management.rst]: @will-moore I will have to defer to @joshmoore for the answer

Comment (joshmoore) [reply docs/group-user-management.rst]: e.g. `omero ldap setdn true --user-name enoether `

Looks like in a refactoring `--group-id` and `--group-name` got added which make no sense here. cc: @sbesson

File: docs/group-user-management.rst (hunk 10 above)
Comment (pwalczysko) [reply docs/group-user-management.rst]: Fixed in https://github.com/ome/omero-guide-introduction/pull/1/commits/b04121dae66a98ac383a0c70b5526ce4beb6fb94

File: docs/group-user-management.rst (hunk 11 above)
Comment (pwalczysko) [reply docs/group-user-management.rst]: > f I did this without the --as-owner flag, would it remove the user-1 from the group?

@will-moore Yes, and the corresponding explanation is added in https://github.com/ome/omero-guide-introduction/pull/1/commits/82a3b9427911c0776b3fd9ef050d43c6133c3e62

File: docs/group-user-management.rst [hunk 12]
```diff
@@ -0,0 +1,253 @@
...
+
+#. Click on the Groups tab. You can search for groups if desired
+
+#. To create a new Group, click on the Add new Group button. Note that the Name and Permissions fields are mandatory.
+
+#. Click Save.
+
+#. To edit a Group, click on the Pencil button |image1|.
+
+#. You can add or remove members or group’s owners or change group permissions. Make sure that the data owned by a user is moved or transferred to another user before removing the user from the group. Having a data owned by someone who is not a member of your group is not desirable.
```
Comment (will-moore) [inline docs/group-user-management.rst]: Maybe after "You can add or remove members or group’s owners or change group permissions." a new bullet point:
"Before removing a user from a group, it is preferable to move data that they own into a different group. Having data in a group where the owner is not a member is not desirable."

File: docs/group-user-management.rst [hunk 13]
```diff
@@ -0,0 +1,253 @@
...
+#. In the top toolbar, click the Admin button |image0|\ . Note that the Admin button is only available for users with certain privileges (administrators and administrators with restricted privileges). If you are a user or a group owner, navigate to the section "Web Interface: Users change their own settings" below.
+
+*Web Interface: Managing Groups*
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+#. Click on the Groups tab. You can search for groups if desired
+
+#. To create a new Group, click on the Add new Group button. Note that the Name and Permissions fields are mandatory.
+
+#. Click Save.
```
Comment (will-moore) [inline docs/group-user-management.rst]: After "Click Save". Add: "The new group will be shown in the list of Groups".
This helps explain the next bullet point better.

File: docs/group-user-management.rst [hunk 14]
```diff
@@ -0,0 +1,253 @@
...
+#. Log in using the username and password provided.
+
+#. In the top toolbar, click the Admin button |image0|\ . Note that the Admin button is only available for users with certain privileges (administrators and administrators with restricted privileges). If you are a user or a group owner, navigate to the section "Web Interface: Users change their own settings" below.
+
+*Web Interface: Managing Groups*
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+#. Click on the Groups tab. You can search for groups if desired
+
+#. To create a new Group, click on the Add new Group button. Note that the Name and Permissions fields are mandatory.
```
Comment (will-moore) [inline docs/group-user-management.rst]: When referring to buttons etc, it's handy to use back-ticks to highlight e.g. ``Add new Group`` button and ``Admin`` button above.

File: docs/group-user-management.rst (hunk 12 above)
Comment (pwalczysko) [reply docs/group-user-management.rst]: Fixed in https://github.com/ome/omero-guide-introduction/pull/1/commits/616ed4142fac38f6c4372be9a497c48f4b5679be and https://github.com/ome/omero-guide-introduction/pull/1/commits/9ef0b8b48f8678d01a9e5963aadc6a9e6b5a29e7

File: docs/group-user-management.rst (hunk 13 above)
Comment (pwalczysko) [reply docs/group-user-management.rst]: Fixed in https://github.com/ome/omero-guide-introduction/pull/1/commits/a184195e748d1a077500817e4ed7e3738d4e39b0

File: docs/group-user-management.rst (hunk 14 above)
Comment (pwalczysko) [reply docs/group-user-management.rst]: Fixed in https://github.com/ome/omero-guide-introduction/pull/1/commits/78470aa5ecab344f13cf03bbbe8d26568400e02f - went for backticks, which are highlighted nicely in red in the final result on readthedoc.

**Comment by pwalczysko** [comment]
All fixes pushed here.

**Comment by jburel** [comment]
small edit pushed. Thanks @pwalczysko

**Pull request merged**